src/renderer/components/inputs/InputContainer.tsx

```diff
@@ -1,64 +1,189 @@
+                            opacity: showHandle ? 1 : 0,
                         },
                         '.react-flow__handle-valid': {
-                            background: '#38A169 !important',
+                            // background: '#38A169 !important',
                         },
                     }}
                 >
-                    <div style={{ position: 'absolute', left: '-4px', width: 0 }}>
-                        <Handle
-                            isConnectable
+                    <Center
+                        left="-4px"
+                        position="absolute"
+                        w={0}
+                    >
+                        <Div
+                            _before={{
+                                content: '" "',
+                                position: 'absolute',
+                                top: '50%',
+                                left: '50%',
+                                height: '30px',
+                                width: '45px',
+                                cursor: 'crosshair',
+                                // backgroundColor: '#FF00FF1F',
+                                transform: 'translate(-50%, -50%)',
+                                borderRadius: '100%',
+                            }}
+                            _hover={{
+                                width: '22px',
+                                height: '22px',
+                                marginLeft: '-3px',
+                            }}
+                            as={LeftHandle}
                             className="input-handle"
                             id={`${id}-${inputId}`}
                             isValidConnection={isValidConnection}
-                            position={Position.Left}
-                            style={{
-                                width: '15px',
-                                height: '15px',
-                                borderWidth: '1px',
-                                borderColor,
-                                transition: '0.25s ease-in-out',
-                                background: handleColor,
+                            sx={{
+                                width: '16px',
+                                height: '16px',
+                                borderWidth: isConnected ? '2px' : '0px',
+                                borderColor: parentTypeColor ?? 'none',
+                                transition: '0.15s ease-in-out',
+                                ...(handleColors.length > 1
+                                    ? { background: isConnected ? connectedColor : handleGradient }
+                                    : {}),
+                                backgroundColor: isConnected ? connectedColor : handleColors[0],
+                                boxShadow: '2px 2px 2px #00000014',
+                                opacity: showHandle ? 1 : 0.5,
```

Comment:
I think going to down to 30% would be even better. And, I think it would be nice if we also applied a grayscale filter. Same for the output container too.

![image](https://user-images.githubusercontent.com/20878432/176404229-6bc68592-7379-43f7-90d3-8e083661b0e0.png)


```suggestion
                                filter: showHandle ? undefined : 'grayscale(100%)',
                                opacity: showHandle ? 1 : 0.3,
```